app/models/coupon.rb

```diff
@@ -41,26 +41,19 @@ class Coupon < ApplicationRecord
   validates :amount_cents, numericality: { greater_than: 0 }
   validates :amount_currency, inclusion: { in: currency_list }
 
-  validates :expiration_duration, numericality: { greater_than: 0 }, if: :time_limit?
-
   scope :order_by_status_and_expiration, lambda {
     order(
       Arel.sql(
         [
           'coupons.status ASC',
           'coupons.expiration ASC',
-          'coupons.created_at + make_interval(days => COALESCE(coupons.expiration_duration, 0)) ASC',
+          'coupons.expiration_date ASC',
         ].join(', '),
       ),
     )
   }
 
-  scope :expired, lambda {
-    where(
-      '(coupons.created_at + make_interval(days => coupons.expiration_duration)) < ?',
-      Time.zone.now.beginning_of_day,
-    )
-  }
+  scope :expired, -> { where('coupons.expiration_date < ?', Time.current.beginning_of_day,) }
```

Comment:
Style/TrailingCommaInArguments: Avoid comma after the last parameter of a method call, unless each item is on its own line. (https://rubystyle.guide#no-trailing-params-comma)